message_verifier_test.go

```diff
@@ -0,0 +1,706 @@
+			requiredHeaders: []string{
+				"(request-target)",
+				"date",
+				"host",
+				"digest",
+			},
+			keyIDMetadataFn: getRSAPublicKey(0, nil),
+			errorExpected:   true,
+		},
+		{
+			title: "multiple signature",
+			req: &http.Request{
+				Method: "POST",
+				URL: &url.URL{
+					Path: "/example",
+				},
+				Header: http.Header{
+					"Date":   {"Fri, 29 Jul 2022 13:23:35 GMT"},
+					"Host":   {"example.org"},
+					"Digest": {"SHA-256=n4bQgYhMfWWaL+qgxVrQFaO/TxsrC4Is0V1sFbDwCgg="},
+				},
+				Body: io.NopCloser(bytes.NewBuffer([]byte("test"))),
+			},
+			signature: `keyId="abc",algorithm="rsa-sha256",headers="(request-target) date host digest",signature="jFpHNN2jW4MTZnZGxm38qx6TqLJCeHEnKW7NhujN6VL7mFbZt6MZ1UcJS/YUYKLnFw1PBbvcOYjTarf2fCtIsCrHKCcYaSaK33L9HrcqoDH/ykOCd57TcNggrFCiqV4sOdjXgFj8Bi7fJbK0QCbIKv03YxBAUiUHArnTP6ANpRiAIyzQxGjGTFjgd2ENgU7Luf346xHdblbwecvJvabmN63mf9a/WFZ21JLERLH2PtAlcbmK43trIjnvHwGeSbi7Pd0xpiXXlhyG+iIdLqbszKjGvYd2cdK4nSfjJhhslA8KS8q5DVp0kJLZKfTbtAVI3mrmVyRAH0ZAa7SXqZQlcQ==",signature="jFpHNN2jW4MTZnZGxm38qx6TqLJCeHEnKW7NhujN6VL7mFbZt6MZ1UcJS/YUYKLnFw1PBbvcOYjTarf2fCtIsCrHKCcYaSaK33L9HrcqoDH/ykOCd57TcNggrFCiqV4sOdjXgFj8Bi7fJbK0QCbIKv03YxBAUiUHArnTP6ANpRiAIyzQxGjGTFjgd2ENgU7Luf346xHdblbwecvJvabmN63mf9a/WFZ21JLERLH2PtAlcbmK43trIjnvHwGeSbi7Pd0xpiXXlhyG+iIdLqbszKjGvYd2cdK4nSfjJhhslA8KS8q5DVp0kJLZKfTbtAVI3mrmVyRAH0ZAa7SXqZQlcQ=="`,
+			requiredHeaders: []string{
+				"(request-target)",
+				"date",
+				"host",
+				"digest",
+			},
+			keyIDMetadataFn: getRSAPublicKey(0, nil),
+			errorExpected:   true,
+		},
+		{
+			title: "unknown algorithm",
+			req: &http.Request{
+				Method: "POST",
+				URL: &url.URL{
+					Path: "/example",
+				},
+				Header: http.Header{
+					"Date":   {"Fri, 29 Jul 2022 13:23:35 GMT"},
+					"Host":   {"example.org"},
+					"Digest": {"SHA-256=n4bQgYhMfWWaL+qgxVrQFaO/TxsrC4Is0V1sFbDwCgg="},
+				},
+				Body: io.NopCloser(bytes.NewBuffer([]byte("test"))),
+			},
+			signature: `keyId="abc",algorithm="ed25519",headers="(request-target) date host digest",signature="jFpHNN2jW4MTZnZGxm38qx6TqLJCeHEnKW7NhujN6VL7mFbZt6MZ1UcJS/YUYKLnFw1PBbvcOYjTarf2fCtIsCrHKCcYaSaK33L9HrcqoDH/ykOCd57TcNggrFCiqV4sOdjXgFj8Bi7fJbK0QCbIKv03YxBAUiUHArnTP6ANpRiAIyzQxGjGTFjgd2ENgU7Luf346xHdblbwecvJvabmN63mf9a/WFZ21JLERLH2PtAlcbmK43trIjnvHwGeSbi7Pd0xpiXXlhyG+iIdLqbszKjGvYd2cdK4nSfjJhhslA8KS8q5DVp0kJLZKfTbtAVI3mrmVyRAH0ZAa7SXqZQlcQ=="`,
+			requiredHeaders: []string{
+				"(request-target)",
+				"date",
+				"host",
+				"digest",
+			},
+			keyIDMetadataFn: getRSAPublicKey(0, nil),
+			errorExpected:   true,
+		},
+		{
+			title: "invalid digest",
```

Comment:
we already have other tests titled `invalid digest`, possibly this should be titled differently. Like `digest does not contain algorithm`